server/routes/private.js

```diff
@@ -50,6 +51,34 @@ router.post('/api/book-appointment', async (req, res) => {
 
 });
 
+router.post('/api/add-recurring-schedule', async (req, res) => {
+    // Validate appropriate parameters are passed into add recurring schedule. 
+    const paramSchema = Joi.object({
+        slotId: Joi.number().integer().required(),
+        schoolId: Joi.number().integer().required(),
+        workerId: Joi.number().integer().required(),
+        status: Joi.string().max(30), // The default value is available which is handled by the database Create Table logic. 
+        date: Joi.date().iso(),
```

Comment:
What happens if no date is specified? Does it input this schedule for all weekdays?

Reply:
@catai9 that was my envision. Is there anything you have in mind? :)

Reply:
Perfect I will add in a comment! :) 
Thank you Amy!

Reply:
It may be better to have a start and an end date to allow for more flexibility in the future. However, for the time being, this is sufficient to allow the user to enter their weekly recurring schedule into the system.